robottelo/ui/repository.py

```diff
@@ -1,45 +1,49 @@
 """Implements Repos UI."""
 from robottelo.constants import CHECKSUM_TYPE, REPO_TYPE
 from robottelo.ui.base import Base, UIError
-from robottelo.ui.locators import common_locators, locators
+from robottelo.ui.locators import common_locators, locators, tab_locators
 from selenium.webdriver.support.select import Select
 
 
 class Repos(Base):
     """Manipulates Repos from UI."""
 
-    def create(self, name, product=None, gpg_key=None, http=False, url=None,
-               upstream_repo_name=None, repo_type=REPO_TYPE['yum'],
+    def navigate_to_entity(self):
+        """Navigate to Repository entity tab"""
+        self.click(tab_locators['prd.tab_repos'])
+
+    def _search_locator(self):
+        """Specify locator for Repository entity search procedure"""
+        return locators['repo.select']
+
+    def create(self, name, product, product_instance, gpg_key=None, http=False,
```

Comment:
`product_instance` is instance of `Product` class. `product` remains the same, and it's actually product name. I can rename it to `product_name` if you want, i just tried to introduce minimum changes, and this product name variable was originally called `product`.